test/es-v5/02.basic-querying.test.js

```diff
@@ -1742,6 +1742,182 @@ describe('basic-querying', function () {
 
   });
 
+  describe('bulk API operations', function(){
+
+    var items = getNewPosts();
+
+    beforeEach(function (done) {
+      this.timeout = 4000;
+      Post.destroyAll(done);
+    });
+
+
+    it('should add some posts at once, using bulkCreate', function(done){
+
+      Post
+        .bulkCreate(items)
+        // NOTE: next step is just a delay, which seems to be required due to a race condition
+        // between bulk operation and find by ID. In result, the following posts array contains
+        // `null` as first item, instead of items[0] from above.
+        .then(delaying)
+        .then(function(ids){
+          ids.should.have.lengthOf(items.length);
+          var gettingPosts = ids.map(function(id){
+            id.should.be.a('string');
+            return Post.findById(id);
+          });
+          return Promise.all(gettingPosts);
+        })
+        .then(function(posts){
+          posts.forEach(function (post, i){
+            post.should.be.an.instanceOf(Post);
+
+            // NOTE: requires because, connector returns arrays as Lists when being a property value
```

Comment:
Can you share an example of arrays vs Lists here?

Reply:
Sure. The following examples are taken from console.log's temporary added to the location in the code you were referring to. Please note, the `comments` filed.
Array:
```js
{ 
  title: 'title 3rd',
  content: 'there is no content. please move on',
  comments: [ 'qwertyid1', 'asdfghid2' ],
  id: 'AV-Msx275g-LBpy2bLHc'
}
```
List:
```js
{ 
  title: 'title 3rd',
  content: 'there is no content. please move on',
  comments: List [ 'qwertyid1', 'asdfghid2' ],
  id: 'AV-Msx275g-LBpy2bLHc'
}
```
I assume it is related to [this type implementation in the juggler](https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/list.js).  I was not able to get it to work with the testing framework. So I decided to do the hack with the serialization-deserialization in order to get a proper array so that the `deep.equal` would pass.